pyInterface/module/utils/_binning.py

```diff
@@ -61,15 +63,94 @@ def __ge__(self, other):
 
 
 	def __str__(self):
-		retval = ""
-		retval += "multiBin: { "
+		retval = "multiBin: " + self.__repr__()
+		return retval
+
+	def __repr__(self):
+		retVariables = []
 		for key in sorted(self.boundaries.keys()):
-			retval += "\"" + key + "\": (" + str(self.boundaries[key][0]) + ", " + str(self.boundaries[key][1]) + "), "
-		retval = retval[:-2] + " }"
+			retVariables.append( "\"" + key + "\": (" + str(self.boundaries[key][0]) + ", " + str(self.boundaries[key][1]) + ")")
+		retval = "{ "
+		retval += ", ".join(retVariables)
+		retval += " }"
 		return retval
 
+	def __hash__(self):
+		keys = tuple(self.boundaries.keys())
+		values = tuple([ self.boundaries[k] for k in keys])
+		return hash((keys, values))
+
+	def __contains__(self, other):
+		'''
+		The other multiBin is contained in this multiBin if the intervals of all variables
+		of this multiBin are larger than the corresponding intervals of the other multiBin.
+		If this multiBin is binned in a variable in which the other multiBin is not binned,
+		the other multiBin is not contained in this multiBin.
+		If the other multiBin is binned in a variable in which this multiBin is not binned,
+		the other multiBin is contained in this multiBin.
+		'''
+		if isinstance(other, dict):
+			other = multiBin(other)
+		if not isinstance(other, multiBin):
+			printErr("Cannot check whether object of type {0} is contained in a multiBin".format(type(other)))
+			exit(1)
+
+		for key in self.boundaries.keys():
+			if key in other.boundaries.keys():
+				if not self.boundaries[key][0] <= other.boundaries[key][0]:
+					return False
+				if not self.boundaries[key][1] >= other.boundaries[key][1]:
+					return False
+			else:
+				return False
+		return True
+
+	def variables(self):
+		'''
+		@return: sorted list of variables defined in the multibin
+		'''
+		return sorted(self.boundaries.keys())
+
+	def uniqueStr(self):
+		'''
+		@return: string, unique for the multibin (without whitespaces)
+		'''
+		out = []
+		for variable in self.variables():
+			variableStr = "{0}_{1!r}_{2!r}".format(variable, self.boundaries[variable][0], self.boundaries[variable][1])
```

Comment:
Personally, I would not use any character here that is allowed in a valid C++ variable name.